community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/LoadCsvAcceptanceTest.scala

```diff
@@ -123,6 +123,28 @@ class LoadCsvAcceptanceTest
     }
   }
 
+  test("import three rows with headers messy data with predicate") {
+    val urls = csvUrls({
+      writer =>
+        writer.println("id,name,x")
+        writer.println("1,'Aardvark',0")
+        writer.println("2,'Babs'")
+        writer.println("3,'Cash',1")
+        writer.println("4,'Dice',\"\"")
+        writer.println("5,'Emerald',")
+    })
+    for (url <- urls) {
+      val result = execute(s"LOAD CSV WITH HEADERS FROM '$url' AS line WITH line WHERE line.name IS NOT NULL RETURN line.x")
```

Comment:
maybe `LOAD CSV ... line.x IS NOT NULL RETURN line.name` is a more meaningful test to have